boot/bootutil/src/boot_record.c

```diff
@@ -23,19 +24,19 @@
 
 #include "mcuboot_config/mcuboot_config.h"
 
+#if defined(__ZEPHYR__) && defined(MCUBOOT_DATA_SHARING)
```

Comment:
Shouldn't this ifdef (without the Zephyr part) just be in Zephyr's mcuboot_config.h? That's where the definitions are needed, anyway.

Reply:
The definitions are only needed for this file (for the version), when included from the mcuboot config file, it causes a lot of build errors, presumably because parts of the code are being setup as "libraries" which would not have access to the application's generated files, which is where the version file would be placed

Reply:
@d3zd3z this is exactly my problem with the proposed solution: Instead of using existing mcuboot solutions the PR needs to overload the existing routines with its own set of routines to get a working solution. This results in os specific parts of the code inside bootutil. The construct is very fragile and will lead to difficult to catch errors. Given the issue #1754 solutions like the proposed should be avoided even if they have shown not to break mcuboot on other oses.
** on top of this the provided solution does not have any benefit over the existing shared memory solution that can easily be defined for zephyr. **

Reply:
I don't really like having this here like this, but I can't really think of a better way of handling it.